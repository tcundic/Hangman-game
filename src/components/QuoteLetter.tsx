import React from "react";

const QuoteLetter = ({letter, isRevealed}: { letter: string, isRevealed: boolean }) => {
    return (
        <div className={`${letter.match(/[^A-Za-z,.?!']/g) ? "column is-0" : "mx-1"}`}>
            {isRevealed ?
                <p>{letter}</p> :
                <p>_</p>
            }
        </div>
    );
};

export default QuoteLetter;