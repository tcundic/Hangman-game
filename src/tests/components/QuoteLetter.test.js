import React from "react";
import { shallow } from "enzyme";
import QuoteLetter from "../../components/QuoteLetter";

test('should render QuoteLetter with hidden letter correctly', () => {
    const wrapper = shallow(<QuoteLetter letter="A" isRevealed={false}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render QuoteLetter with displayed letter correctly', () => {
    const wrapper = shallow(<QuoteLetter letter="A" isRevealed={true}/>);
    expect(wrapper).toMatchSnapshot();
});
