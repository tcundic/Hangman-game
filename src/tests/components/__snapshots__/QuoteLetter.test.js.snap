// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render QuoteLetter with displayed letter correctly 1`] = `
<div
  className="mx-1 is-inline-block-touch"
>
  <p>
    A
  </p>
</div>
`;

exports[`should render QuoteLetter with hidden letter correctly 1`] = `
<div
  className="mx-1 is-inline-block-touch"
>
  <p>
    _
  </p>
</div>
`;
